### PR TITLE
kustomize-sops: rename exec plugin to ksops

### PR DESCRIPTION
**Pull request by starcraft66**
###### Description of changes

Addresses https://github.com/NixOS/nixpkgs/issues/175515

Renames exec plugin from `ksops-exec` to `ksops` like it was done upstream in v3.
Creates a symlink from `ksops-exec` to `ksops` for the sake of backwards compatibility.

```
/nix/store/5kkrsfvl5pp1cgk0mw1mp7smq1pr011c-kustomize-sops-3.0.2/
└── lib/
   └── viaduct.ai/
      └── v1/
         ├── ksops/
         │  └── ksops*
         └── ksops-exec/
            └── ksops-exec -> ../ksops/ksops*
```

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SnO2WMaN** [comment]
@starcraft66 

[The original repository](https://github.com/viaduct-ai/kustomize-sops) generates executable file named `ksops`, so I think the attempt to rename is good.

However, the current pull request (and the conventional configuration) put the executable in `$out/lib/viaduct.ai/v1/ksops/`, so it cannot be used in the nix shell because the path does not pass through.

This is not the purpose of this PR, but simply remove `installPhase` (in this case, the executable file can be used as `kustomize-sops`), or place it in `$out/bin` or some other appropriate location.

I tested here and it should probably work.

https://github.com/SnO2WMaN/nixpkgs/commit/9ab7594843ae4ebee07fed85bb133e1cf69d3bb2#diff-e8d0d050c71879a60e257a7c790692381f4a6b360e1bf13b0c3430bb41baa96c.

**Comment by starcraft66** [comment]
> I tested here and it should probably work.

Not sure what you are trying to do here, that would break kustomize. An example of the correct way to use this package in a nix-shell can be found here: https://github.com/starcraft66/infrastructure/blob/master/flake.nix#L44-L54

**Comment by SnO2WMaN** [comment]
I'm sorry, I was very much mistaken about ksops. My apologies.

**Comment by plaflamme** [comment]
@starcraft66 @kalbasit is anything holding this back? According to their README, the binary is now expected to be on the `PATH` as `ksops` directly ([here](https://github.com/viaduct-ai/kustomize-sops?tab=readme-ov-file#sanity-checks))

**Comment by starcraft66** [comment]
A review is what's always been holding me back

**Review by ProjectInitiative — APPROVED** [review]
Tested on linux-x86.
`nixpkgs-review pr 175539` came back successful. Not sure if any other maintainers can review this to get the changes merged.

**Review by kalbasit — APPROVED** [review]

**Comment by kalbasit** [comment]
hey sorry for the hold-up y'all, I somehow missed the notification on this pr, apologies. LGTM.

**Pull request merged**
